src/pages/postList/PostForms/GeneralPostForm.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+// import form data interface
+import { IFormInputs } from "./IFormInputs";
+// import reusable form input components
+import {
+    SelectInput,
+    TagsInput,
+    TextFieldInput,
+} from "../../../components/FormInput";
+
+// Import MUI components
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Stack from "@mui/material/Stack";
+
+// Define Yup validation schema, will be updated later
+const validationSchema = yup.object().shape({
+    category: yup.string().required(),
+    title: yup.string().required(),
+    content: yup.string().required(),
+    tags: yup.array().required(),
+});
+interface Props {
+    onSubmit: SubmitHandler<IFormInputs>;
+}
+
+const GeneralPostForm = ({ onSubmit }: Props) => {
+    // tags and categories, (will be updated later)
+    const tagOptions = ["frontend", "backend", "fullstack", "database"];
+    const categoryOptions = ["learn", "interview", "project", "general"];
+    // react hook form
+    const {
+        control,
+        handleSubmit,
+        formState: { errors },
+    } = useForm<IFormInputs>({
+        // mode: "onChange",
+        resolver: yupResolver(validationSchema),
+    });
+    return (
+        <Box component="form" onSubmit={handleSubmit(onSubmit)}>
+            {/* post category */}
+            <SelectInput
+                control={control}
+                error={errors.category}
+                helperText="Choose a category for your post"
+                name="category"
+                options={categoryOptions}
+            />
+            {/* post title */}
+            <TextFieldInput
+                control={control}
+                error={errors.title}
+                helperText="Post title"
+                name="title"
+                rows={1}
+            />
+            {/* post content                    */}
+            <TextFieldInput
+                control={control}
+                error={errors.content}
+                helperText="Post content"
+                name="content"
+                rows={8}
+            />
```

Comment:
I recommend adding a rich-text editor for a better experience

Reply:
It is a good idea. Just like the toolbar of this comment editor, If we allow a rich-text editor, each post might have different styles. We need to consider what styles can be edited by users.  I will add it to our next meeting topics.

Reply:
Do you know what format rich text editors output content in? We need to make sure it can easily be stored in the database, processed etc.

Reply:
> Do you know what format rich text editors output content in?
Usually HTML5 but some are XHTML

This one especially in HTML:
[React Quill](https://zenoamaro.github.io/react-quill/)

Reply:
Ok, so we will need to be aware of possible SQL injection attacks. I believe React automatically escapes HTML entities to help prevent this but we would need to double check that that would be enough to ensure that our application is secure. It's a good idea. Modern and still only really found on better designed websites